target/linux/ramips/dts/mt7628an_tplink_tl-wa850re-v4.dts

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,tl-wa850re-v4", "mediatek,mt7628an-soc";
+	model = "TP-Link TL-WA850RE v4";
+
+	chosen {
+		bootargs = "console=ttyS0,57600n8";
+	};
+
+	aliases {
+		label-mac-device = &ethernet;
+
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "tl-wa850re-v4:blue:power";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "tl-wa850re-v4:blue:lan";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "tl-wa850re-v4:blue:wlan";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		range_extender {
+			label = "tl-wa850re-v4:blue:re";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi1 {
+			label = "tl-wa850re-v4:blue:rssi1";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi2 {
+			label = "tl-wa850re-v4:blue:rssi2";
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi3 {
+			label = "tl-wa850re-v4:blue:rssi3";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi4 {
+			label = "tl-wa850re-v4:blue:rssi4";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi5 {
+			label = "tl-wa850re-v4:blue:rssi5";
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x20000 0x3a0000>;
+			};
+
+			tpconfig: partition@3c0000 {
+				label = "config";
+				reg = <0x3c0000 0x30000>;
+				read-only;
+			};
+
+			/* or radio, but mt2628an.dtsi likes factory */
+			factory: partition@3f0000 {
```

Comment:
You may call it radio as well, but please remove the comment.

Reply:
Removed comment, left as factory.

Reply:
Use radio:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/mt7628an_tplink_re305-v1.dts#L109

factory would imply the use of specific MAC address locations etc., and that's not what TP-Link does here.

Reply:
Ah sorry i missed this comment previously - i'll change it to radio (and add the "mediatek,mtd-eeprom = <&radio 0x0>" line... it was so convenient that it defaulted to factory 0x0 :P)